gcc/rust/hir/rust-ast-lower-stmt.h

```diff
@@ -445,6 +445,19 @@ class ASTLoweringStmt : public ASTLoweringBase
     translated = fn;
   }
 
+  void visit (AST::MacroRulesDefinition &macro_def) override
+  {
+    // FIXME: This lowers macro definitions as statements (i.e. macros defined
```

Comment:
good comment. Do we need this in the item scope? I guess it must be more permissive there.

Reply:
It is more permissive in the item scope, the crate visitor does not check if an item has "not" been lowered to HIR. Block on the other hand will error out if one of their statements gets lowered to a null pointer.